src/components/protocol_handler/src/protocol_handler_impl.cc

```diff
@@ -2010,12 +2031,20 @@ void ProtocolHandlerImpl::NotifySessionStarted(
     }
 
     if (!rejected_params.empty()) {
+      service_status_update_handler_->OnServiceUpdate(
+          connection_key,
+          context.service_type_,
+          ServiceStatus::SERVICE_START_FAILED);
       SendStartSessionNAck(context.connection_id_,
                            packet->session_id(),
                            protocol_version,
                            packet->service_type(),
                            rejected_params,
-                           "SSL Handshake failed due to rejected parameters");
+                           "SSL Handshake failed due to rejected parameters",
+                           full_version);
+      if (packet->service_type() != kRpc) {
```

Comment:
@VladSemenyuk why only audio/video?

Reply:
We shoudn't delete session after unsuccessful start rpc service ack for protected mode